### PR TITLE
valign HTML to vertical-align CSS

### LINKED ISSUE #2798: Fix HTML Validator errors for templates
Child issue of #163. Aims to resolve all HTML validation issues associated with templates.

### Related files
Folders
- [ ] account
- [ ] authors
- [ ] books
- [ ] contact
- [ ] covers
- [ ] email
- [ ] history
- [ ] home
- [ ] languages
- [ ] lib
- [ ] libraries
- [ ] lists
- [ ] merge	
- [ ] publishers
- [ ] recentchanges
- [ ] search
- [ ] site	
- [ ] stats	
- [ ] subjects	
- [ ] type	
- [ ] work

Files
- [x] account.html
- [x] borrow_admin.html 
- [ ] borrow_admin_no_update.html [IN PROGRESS]
- [x] design.html	
- [ ] diff.html	
- [ ] edit_yaml.html	
- [ ] editpage.html	
- [ ] form.html	
- [ ] history.html	
- [ ] internalerror.html	
- [ ] login.html
- [ ] message.html
- [ ] messages.html
- [ ] notfound.html	
- [ ] permission.html	
- [ ] permission_denied.html	
- [ ] showamazon.html	
- [ ] showia.html	
- [ ] showmarc.html	
- [ ] site.html	
- [ ] status.html	
- [x] subjects.html
- [ ] support.html	
- [ ] viewpage.html	
- [ ] widget.html	
- [ ] work_search.html	

### Stakeholders
@jdlrobson @mekarpeles @koderjoker

### PR DESCRIPTION
**Pull request by ArunTeltia**
Partially resolve #2798

Replace HTML valign with CSS vertical-align

### Stakeholders
@jdlrobson @mekarpeles

### CODE REVIEW

**Comment by ArunTeltia** [comment]
@jdlrobson I tried to find the CSS file for both the file but I am not able to find it can you help me where we need to change the CSS

**Comment by jdlrobson** [comment]
What is the URL of the page is this HTML used on?

Finding out where the CSS lives is the "meat" of this task and the work involved to be able to fix it but it will be somewhere in https://github.com/internetarchive/openlibrary/tree/master/static/css

**Comment by ArunTeltia** [comment]
Please check @jdlrobson now

File: openlibrary/catalog/edition_merge/templates/index.html [hunk 1]
```diff
@@ -1,61 +1,69 @@
...
+    };
+  </style>
 </head>
+
 <body>
-<h1>Merge duplicate editions</h1>
+  <h1>Merge duplicate editions</h1>
 
-{% with messages = get_flashed_messages() %}
+  {% with messages = get_flashed_messages() %}
```
Comment (jdlrobson) [inline openlibrary/catalog/edition_merge/templates/index.html:16]: These indentation changes don't seem related. Can you explain what they are for in the pull request description or remove them?

File: openlibrary/templates/admin/memory/index.html [hunk 2]
```diff
@@ -13,7 +13,7 @@ <h1>Memory Status</h1>
         margin: 20px 0px;
     }
     td{
-        valign: top;
+        vertical-align: top;
```
Comment (jdlrobson) [inline openlibrary/templates/admin/memory/index.html:16]: great catch!

File: static/css/legacy.less [hunk 3]
```diff
@@ -1416,6 +1416,12 @@ tr {
       padding-left: 30px !important;
     }
   }
+  td.valign_mid{
```
Comment (jdlrobson) [inline static/css/legacy.less:1419]: The `td` makes these overly specific - I think `.valign_mid` should do. If not please add an inline comment explaining why they are necessary.

File: static/css/legacy.less [hunk 4]
```diff
@@ -1416,6 +1416,12 @@ tr {
...
+    vertical-align:"middle"
+  }
+  td.valign_top{
```
Comment (jdlrobson) [inline static/css/legacy.less:1422]: I think this is a step in the right direction but I suggest calling this class .column--aligned-top to be consistent with the other BEM naming we've been using recently.

File: static/css/legacy.less (hunk 3 above)
Comment (jdlrobson) [reply static/css/legacy.less:1419]: I think this is a step in the right direction but I suggest calling this class .column--aligned-middle to be consistent with the other BEM naming we've been using recently.

File: openlibrary/catalog/edition_merge/templates/index.html [hunk 5]
```diff
@@ -1,61 +1,69 @@
 <html>
+
 <head>
-<title>Merge duplicate editions</title>
-<style>
-td { vertical-align: top; background: #ddd };
-</style>
+  <title>Merge duplicate editions</title>
+  <style>
+    td {
```
Comment (jdlrobson) [inline openlibrary/catalog/edition_merge/templates/index.html:6]: Why is this needed if its also in the legacy stylesheet?

File: openlibrary/catalog/edition_merge/templates/index.html [hunk 6]
```diff
@@ -1,61 +1,69 @@
 <html>
+
 <head>
-<title>Merge duplicate editions</title>
```
Comment (jdlrobson) [inline openlibrary/catalog/edition_merge/templates/index.html:3]: These whitespace changes seem unrelated and make this pull request harder to read and understand. Can you remove?

File: openlibrary/catalog/edition_merge/templates/index.html [hunk 7]
```diff
@@ -1,61 +1,69 @@
...
-<td>
-{% endif %}
-{{ unmerge_count }}
-</td>
-</tr>
-{% endfor %}
-</table>
-
-{{ pager() }}
+  {% endwith %}
```
Comment (jdlrobson) [inline openlibrary/catalog/edition_merge/templates/index.html:24]: Why so many changes?

**Review by jdlrobson — CHANGES_REQUESTED** [review]
I think youv'e found the right place for the styles but the changes to the HTML should be confined to the class name.

The indentation changes make this pull request impossible to review, but I think if you can revise this to the minimum needed we have a solution here!

**Comment by ArunTeltia** [comment]
Yes I am working on these changes that you asked for right now thank you for telling me 
I think I got confused and indent it using prettier, I will revert these

File: static/css/legacy.less (hunk 3 above)
Comment (ArunTeltia) [reply static/css/legacy.less:1419]: Okay I will change these

**Comment by ArunTeltia** [comment]
I have updated as per your request @jdlrobson please review it

File: openlibrary/catalog/edition_merge/templates/index.html [hunk 8]
```diff
@@ -32,7 +32,7 @@ <h1>Merge duplicate editions</h1>
 
 {{ pager() }}
 
-{{ total }} items need to be merged ({{ easy }} easy)<p>
+{{ total }} items need to be merged ({{ easy }} easy)</p>
```
Comment (jdlrobson) [inline openlibrary/catalog/edition_merge/templates/index.html:35]: Where is the opening p tag? 
It looks like the <p> inside the macro pager() is illplaced to me. Should that be on this line as well?

e.g. `<p> {{ total }} items need to be merged ({{ easy }} easy)</p>`

**Review by jdlrobson — COMMENTED** [review]
Thank you for cleaning this up! This looks much clearer now!

 I just need to find time to test this now, but just one comment here before I do that.

File: openlibrary/catalog/edition_merge/templates/index.html (hunk 8 above)
Comment (ArunTeltia) [reply openlibrary/catalog/edition_merge/templates/index.html:35]:
```
<p>
{% endmacro %}

{{ pager() }}

{{ total }} items need to be merged ({{ easy }} easy)<p>
```
line 2 in https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/edition_merge/templates/index.html#L28
the p tag is opened at line 30 but not closed anywhere it was reopened 
so I closed it not a scope for this pr but I found it and replaced it :D

Comment (ArunTeltia) [reply openlibrary/catalog/edition_merge/templates/index.html:35]: I guess the p tag at line 30 is illegal and it must be this 
```<p> {{ total }} items need to be merged ({{ easy }} easy)</p>```
as you said 
I am thinking of reverting this back I will create a different pr for this

File: static/css/legacy.less [hunk 9]
```diff
@@ -1416,6 +1416,12 @@ tr {
       padding-left: 30px !important;
     }
   }
+  .column--aligned-middle{
+    vertical-align:"middle"
```
Comment (jdlrobson) [inline static/css/legacy.less:1420]: no quotes on CSS properties.
`vertical-align:middle;` will work fine

Comment (ArunTeltia) [reply static/css/legacy.less]: I had resolved this issue, can you check this now

**Comment by ArunTeltia** [comment]
Can you please check now @jdlrobson 
and I thank you for being supportive through out this issue :D

File: openlibrary/catalog/edition_merge/templates/index.html [hunk 10]
```diff
@@ -58,4 +58,4 @@ <h1>Merge duplicate editions</h1>
 {{ pager() }}
 
 </body>
-</html>
+</html>
```
Comment (jdlrobson) [inline openlibrary/catalog/edition_merge/templates/index.html:61]: unrelated whitespace change here. I can amend on merge if necessary.

**Comment by jdlrobson** [comment]
There are a few errors in Travel CI that need attention first relating to the new bundle size (`npm test`):
https://github.com/internetarchive/openlibrary/pull/3976/checks?check_run_id=1349596165

Could you also share which URLs you are using locally for testing?

File: openlibrary/catalog/edition_merge/templates/index.html (hunk 10 above)
Comment (ArunTeltia) [reply openlibrary/catalog/edition_merge/templates/index.html]: fixed this

**Comment by ArunTeltia** [comment]
sorry for the late message actually I was looking into why this is failing the test now 
I am sorry I don't understand which URL you are talking about :(
In the test, it was showing there is some problem related to some size of the bundle 
Is there a good way to resolve this

**Comment by jdlrobson** [comment]
> I am sorry I don't understand which URL you are talking about :(

How are you testing the changes locally? I assume you are using pages on localhost:8080 to verify your patch fixes these problems and I would like to also test these pages myself.


> In the test, it was showing there is some problem related to some size of the bundle
Is there a good way to resolve this

The bundle size test can be run with `npm test` please update /bundlesize.config.json rounding to the nearest 0.1kb.

**Comment by ArunTeltia** [comment]
For locally testing, I have established the docker environment, and I refresh every time I make some changes

**Comment by ArunTeltia** [comment]
okay I am updating the /bundlesize.config.json

**Comment by jdlrobson** [comment]
> For locally testing, I have established the docker environment, and I refresh every time I make some changes

The pages you are changing have specific URLs. For example, the admin/memory.html is the template for http://localhost:8080/admin/memory. The homepage is http://localhost:8080/ but that is not impacted by the changes here so refreshing that is not suitable for testing anything here.

You are changing at least 2 more pages - the edition merge page and a page that uses "EditionAuthorRow.html". Which URLs correspond to pages using those?

**Comment by ArunTeltia** [comment]
The URL is /book/add 
The file is /openlibrary/admin/numbers.py/admin_total__edition
and after changing the /bundlesize.config.json from 20 to 21 the book the Travis ci test case passes

**Comment by jdlrobson** [comment]
> The URL is /book/add

I get a 404 on http://localhost:8080/book/add - is that a typo?

**Review by jdlrobson — COMMENTED** [review]
While I can confirm the fix  for  the changes in openlibrary/templates/admin/memory/index.html 

I'm looking through the code and I cannot see any usages of LanguageTable and EditionAuthorRow anywhere and it looks like dead code. What am I missing?

**Comment by ArunTeltia** [comment]
Really sorry about that its /books/add not book/add
Its 5am in my country so I guess I am sleepy :P

**Comment by jdlrobson** [comment]
None of these templates are being used on http://localhost:8080/books/add from what I can see... so I a little confused. Let's pick this up tomorrow.

**Comment by ArunTeltia** [comment]
> I'm looking through the code and I cannot see any usages of LanguageTable and EditionAuthorRow anywhere and it looks like dead code. What am I missing?
I actually stumble on this when i am trying to find problems by going to each HTML pages and checking for any syntax error that's where i found this

**Comment by ArunTeltia** [comment]
> None of these templates are being used on http://localhost:8080/books/add from what I can see... so I a little confused. Let's pick this up tomorrow.

okay :D

**Comment by jdlrobson** [comment]
> I actually stumble on this when i am trying to find problems by going to each HTML pages and checking for any syntax error that's where i found this

How are you going to these pages and checking them? The templates themselves, despite the html file extension are not HTML pages. They are templates that may or may nor create HTML.  
If you want to submit a pull request for the change to openlibrary/templates/admin/memory/index.html  I can verify that and will be happy to merge that straightaway. This template as I've pointed out modifies http://localhost:8080/admin/memory and when I put the output HTML through https://validator.w3.org/ it validates fine.

Looking through the rest of your changes, I'm pretty sure its better to remove the  openlibrary/macros/EditionAuthorRow.html  and  openlibrary/macros/LanguageTable.html  files. The newly added CSS is not necessary as these macros are not used in any openlibrary pages and their existence causes confusion such as what we have here. We should confirm @mekarpeles that these are not used and have those removed rather than modified.

Next steps:
* please send a new pull request for the openlibrary/templates/admin/memory/index.html change.
* Let's wait on some feedback from Mek amend this patchset to remove the files in question if he can confirm.

**Comment by ArunTeltia** [comment]
I am not sure if i am able to find any macros file in open library, and I am checking ```http://localhost:8080/admin/memory```  now :D

**Comment by ArunTeltia** [comment]
Summary for @mekarpeles 
I have worked on changing valign to vertical-align in the entire project but as I proceed @jdlrobson found that some pages are not being used in openlibrary now 

> Looking through the rest of your changes, I'm pretty sure its better to remove the openlibrary/macros/EditionAuthorRow.html and openlibrary/macros/LanguageTable.html files. The newly added CSS is not necessary as these macros are not used in any openlibrary pages and their existence causes confusion such as what we have here. We should confirm @mekarpeles that these are not used and have those removed rather than modified.

File: static/css/legacy.less (hunk 4 above)
Comment (cdrini) [reply static/css/legacy.less]: DONE

File: static/css/legacy.less (hunk 3 above)
Comment (cdrini) [reply static/css/legacy.less]: DONE

File: openlibrary/catalog/edition_merge/templates/index.html (hunk 5 above)
Comment (cdrini) [reply openlibrary/catalog/edition_merge/templates/index.html]: DONE

File: openlibrary/catalog/edition_merge/templates/index.html (hunk 6 above)
Comment (cdrini) [reply openlibrary/catalog/edition_merge/templates/index.html]: DONE

File: openlibrary/catalog/edition_merge/templates/index.html (hunk 7 above)
Comment (cdrini) [reply openlibrary/catalog/edition_merge/templates/index.html]: DONE

**Review by cdrini — CHANGES_REQUESTED** [review]
Hi @ArunTeltia ! @jdlrobson is correct, it looks like both `EditionAuthorRow.html` and `LanguageTable.html` are unused. I confirmed that they have no other references. Let's remove them entirely :)

**Comment by cdrini** [comment]
Oh! Also: Remove the `package-lock.json` from your changes (i.e.

```
git checkout master
git pull upstream master
git checkout valign
git checkout master package-lock.json
```

). There shouldn't be any changes to that file as a result of these changes.
